Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -3969,10 +3969,14 @@ JSC_DEFINE_JIT_OPERATION(operationArrayIncludesValueInt32, UCPUStrictInt32, (JSG
     if (searchElement.isUndefined() && containsHole(data, length))
         OPERATION_RETURN(scope, toUCPUStrictInt32(1));
 
-    if (!searchElement.isInt32AsAnyInt())
-        OPERATION_RETURN(scope, toUCPUStrictInt32(0));
+    bool searchingNegZero = false;
+    if (!searchElement.isInt32AsAnyInt()) {
+        searchingNegZero = searchElement.isNumber() && searchElement.asNumber() == -0;
```

Comment:
`== -0` does not return `false` for positive zero. So, just using `== 0` is better for readability. (Using `-0` here is misleading since it looks like we are actually checking `-0`, but it is not).